frontend/src/metabase/admin/people/hooks/use-people-query.js

```diff
@@ -15,20 +15,28 @@ export const usePeopleQuery = pageSize => {
   const { handleNextPage, handlePreviousPage, setPage, page } = usePagination();
 
   const [status, setStatus] = useState(USER_STATUS.active);
-  const [searchInputValue, setSearchInputValue] = useState("");
+  const [searchInputValue, _setSearchInputValue] = useState("");
+  const dirtyRef = useRef(false);
+  const setSearchInputValue = useCallback(setter => {
+    _setSearchInputValue(setter);
+    dirtyRef.current = true;
+  }, []);
 
   const [searchText, setSearchText] = useState("");
 
   useEffect(() => {
-    const timerId = setTimeout(() => {
-      const searchText =
-        searchInputValue.length >= MIN_SEARCH_LENGTH ? searchInputValue : "";
+    // skip running if the user has not changed any input
```

Comment:
Bug where pagination would always get set to page 0 `SEARCH_DEBOUNCE_DURATION` after mount. This was causing a race condition in our e2e where ~20% we'd flake from the test expecting to be on another page but it had gotten reset.